temporal-sdk/src/main/java/io/temporal/worker/tuning/ResourceBasedSlotOptions.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (C) 2022 Temporal Technologies, Inc. All Rights Reserved.
+ *
+ * Copyright (C) 2012-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Modifications copyright (C) 2017 Uber Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this material except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.temporal.worker.tuning;
+
+import io.temporal.common.Experimental;
+import java.time.Duration;
+
+/** Options resource-based slot suppliers */
+@Experimental
+public class ResourceBasedSlotOptions {
```

Comment:
There is a bit of a mix between builder patterns and POJOs for options that is a bit inconsistent. Maybe it's ok, but just threw me off a bit.

Reply:
Yeah I think all classes called `*Options`  in the Java SDK follow a builder  pattern

Reply:
I think we should make this a builder and add all the boilerplate helpers like all the other options classes have